x-pack/plugins/security_solution/public/management/pages/trusted_apps/service/trusted_apps_http_service.ts

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import pMap from 'p-map';
+import { HttpStart } from 'kibana/public';
+import {
+  ENDPOINT_TRUSTED_APPS_LIST_ID,
+  EXCEPTION_LIST_ITEM_URL,
+  EXCEPTION_LIST_URL,
+} from '@kbn/securitysolution-list-constants';
+import {
+  ExceptionListItemSchema,
+  ExceptionListSummarySchema,
+  FoundExceptionListItemSchema,
+} from '@kbn/securitysolution-io-ts-list-types';
+import {
+  DeleteTrustedAppsRequestParams,
+  GetOneTrustedAppRequestParams,
+  GetOneTrustedAppResponse,
+  GetTrustedAppsListRequest,
+  GetTrustedAppsListResponse,
+  GetTrustedAppsSummaryRequest,
+  MaybeImmutable,
+  PostTrustedAppCreateRequest,
+  PostTrustedAppCreateResponse,
+  PutTrustedAppsRequestParams,
+  PutTrustedAppUpdateRequest,
+  PutTrustedAppUpdateResponse,
+  TrustedApp,
+} from '../../../../../common/endpoint/types';
+import { sendGetEndpointSpecificPackagePolicies } from '../../policy/store/services/ingest';
+import { isGlobalEffectScope } from '../state/type_guards';
+import { toUpdateTrustedApp } from '../../../../../common/endpoint/service/trusted_apps/to_update_trusted_app';
+import { validateTrustedAppHttpRequestBody } from './validate_trusted_app_http_request_body';
+import {
+  exceptionListItemToTrustedApp,
+  newTrustedAppToCreateExceptionListItem,
+  updatedTrustedAppToUpdateExceptionListItem,
+} from './mappers';
+
+export interface TrustedAppsService {
+  getTrustedApp(params: GetOneTrustedAppRequestParams): Promise<GetOneTrustedAppResponse>;
+
+  getTrustedAppsList(request: GetTrustedAppsListRequest): Promise<GetTrustedAppsListResponse>;
+
+  deleteTrustedApp(request: DeleteTrustedAppsRequestParams): Promise<void>;
+
+  createTrustedApp(request: PostTrustedAppCreateRequest): Promise<PostTrustedAppCreateResponse>;
+
+  updateTrustedApp(
+    params: PutTrustedAppsRequestParams,
+    request: PutTrustedAppUpdateRequest
+  ): Promise<PutTrustedAppUpdateResponse>;
+
+  getPolicyList(
+    options?: Parameters<typeof sendGetEndpointSpecificPackagePolicies>[1]
+  ): ReturnType<typeof sendGetEndpointSpecificPackagePolicies>;
+
+  assignPolicyToTrustedApps(
+    policyId: string,
+    trustedApps: MaybeImmutable<TrustedApp[]>
+  ): Promise<PutTrustedAppUpdateResponse[]>;
+
+  removePolicyFromTrustedApps(
+    policyId: string,
+    trustedApps: MaybeImmutable<TrustedApp[]>
+  ): Promise<PutTrustedAppUpdateResponse[]>;
+}
+
```

Comment:
I'm running out of time for 8.0. For 8.1, it would be good to do that - so as you and @academo work on per-policy event filters + host isolation exceptions, that would be a better time (IMO 😄 ).

is that cool?

Reply:
it makes sense to me. Focus on using 8.0 to deprecate the API and 8.1 to normalize it.

Reply:
In complete agreement :)